packages/image_picker/android/src/main/java/io/flutter/plugins/imagepicker/ImagePickerDelegate.java

```diff
@@ -95,81 +100,88 @@
     void onPathReady(String path);
   }
 
-  @VisibleForTesting final String fileProviderName;
-
-  @VisibleForTesting
-  PermissionManager permissionManager =
-      new PermissionManager() {
-        @Override
-        public boolean isPermissionGranted(String permissionName) {
-          return ActivityCompat.checkSelfPermission(activity, permissionName)
-              == PackageManager.PERMISSION_GRANTED;
-        }
-
-        @Override
-        public void askForPermission(String permissionName, int requestCode) {
-          ActivityCompat.requestPermissions(activity, new String[] {permissionName}, requestCode);
-        }
-      };
-
-  @VisibleForTesting
-  IntentResolver intentResolver =
-      new IntentResolver() {
-        @Override
-        public boolean resolveActivity(Intent intent) {
-          return intent.resolveActivity(activity.getPackageManager()) != null;
-        }
-      };
-
-  @VisibleForTesting
-  FileUriResolver fileUriResolver =
-      new FileUriResolver() {
-        @Override
-        public Uri resolveFileProviderUriForFile(File file) {
-          return FileProvider.getUriForFile(activity, fileProviderName, file);
-        }
-
-        @Override
-        public void getFullImagePath(final OnPathReadyListener listener) {
-          MediaScannerConnection.scanFile(
-              activity,
-              new String[] {pendingCameraImageUri.getPath()},
-              null,
-              new MediaScannerConnection.OnScanCompletedListener() {
-                @Override
-                public void onScanCompleted(String path, Uri uri) {
-                  listener.onPathReady(path);
-                }
-              });
-        }
-      };
-
-  @VisibleForTesting FileUtils fileUtils = new FileUtils();
-
   private Uri pendingCameraImageUri;
   private MethodChannel.Result pendingResult;
   private MethodCall methodCall;
 
   public ImagePickerDelegate(
-      Activity activity, File externalFilesDirectory, ImageResizer imageResizer) {
+      final Activity activity, File externalFilesDirectory, ImageResizer imageResizer) {
     this.activity = activity;
```

Comment:
Instead of duplicating the initialization of the instance variables, could this constructor just new up the PermissionManager, IntentResolver, FileUriResolver, and FileUtiils and then pass those in addition to the other arguments to the other constructor? For a simple example of what I mean, check this out: https://stackoverflow.com/questions/18935559/java-calling-constructor-within-another

Reply:
This proved to be hard. 

Well, not hard, but would require refactoring the `FileUriResolver` et al to not reference any fields from the `ImagePickerDelegate` class at all. Which would make for a slightly uglier code. I can do it, but I estimate it would take at least an hour; I would try to see if it would make some of those into separate class(es?) and do thorough manual testing after that.

If it's still needed, I can still work on this maybe later this week.

Reply:
How's the `FileUriResolver` using fields from `ImagePickerDelegate`? Do the values for `activity`, etc not get closured in from the local context where the `FileUriResolver` class is created?

Reply:
You're right. Seems like it was using only `fileProviderName` and `pendingCameraImageUri` fields. I'll see if I can resolve this right now.

Reply:
I was able to make the changes, but they broke a couple tests. I'll have to modify the tests, make sure that they pass and still cover everything I originally intended them to do. After that, I'll manually verify that everything works fine and changing behavior on the production code makes the tests fail.

Since I already did one of those rounds tonight and it's getting pretty late here in Finland, I'll get back to this tomorrow.

Reply:
Sounds good. Thanks for working on it. Good night!